spec/symengine_spec.rb

```diff
@@ -18,5 +18,52 @@
     it { is_expected.to be_a SymEngine::Rational }
     its(:to_s) { is_expected.to eq '1/3' }
   end
+  
+  describe 'parse' do
+    subject { SymEngine::parse('123 + 321') }
+    
+    it { is_expected.to be_a SymEngine::Integer }
+    it { is_expected.to eq 444 }
+  end
+
+  it 'gives parse errors' do
+    expect { SymEngine::parse('12a + n34a9') }.to raise_error(RuntimeError)
```

Comment:
Found just the thing that explains creating custom exceptions from Ruby C API: http://clalance.blogspot.in/2011/01/writing-ruby-extensions-in-c-part-5.html